src/status_im/transport/inbox.cljs

```diff
@@ -136,69 +157,110 @@
+  [db]
+  (conj (mapv :topic
+              (remove :fetch-history?
+                      (vals (:transport/chats db))))
+        (transport.utils/get-topic constants/contact-discovery)))
+
+(defn get-request-history-topics
+  "Returns topics for which full history has not been recovered"
+  [db]
+  (mapv :topic
+        (filter :fetch-history?
+                (vals (:transport/chats db)))))
 
 (defn request-messages
-  ([cofx]
-   (request-messages {} cofx))
-  ([{:keys [topics discover? should-recover?]
-     :or {should-recover? true
-          discover?       true}}
-    {:keys [db] :as cofx}]
-   (let [mailserver-status (:mailserver-status db)
-         sym-key-id        (:inbox/sym-key-id db)
-         wnode             (get-current-wnode-address db)
-         inbox-topics      (get-topics db topics discover?)
-         inbox-ready?      (and (= :connected mailserver-status)
-                                sym-key-id)]
-     (when should-recover?
-       (if inbox-ready?
-         {::request-messages {:wnode      wnode
-                              :topics     (into [] inbox-topics)
-                              :sym-key-id sym-key-id
-                              :web3       (:web3 db)}
-          :db                (assoc db
-                                    :inbox/fetching? true
-                                    :inbox/topics #{})
-          :dispatch-later    [{:ms fetching-messages-notification-timeout
-                               :dispatch [:inbox/remove-fetching-notification]}]}
-         {:db (assoc db :inbox/topics (into #{} inbox-topics))})))))
+  ([{:keys [db now] :as cofx}]
+   (let [wnode                   (get-current-wnode-address db)
+         web3                    (:web3 db)
+         sym-key-id              (:inbox/sym-key-id db)
+         now-in-s                (quot now 1000)
+         last-request            (get-in db [:account/account :last-request]
+                                         (- now-in-s (* 3600 24)))
+         request-messages-topics (get-request-messages-topics db)
+         request-history-topics  (get-request-history-topics db)]
+     (when (inbox-ready? cofx)
+       {::request-messages {:wnode      wnode
+                            :topics     request-messages-topics
+                            :from       last-request
+                            :to         now-in-s
+                            :sym-key-id sym-key-id
+                            :web3       web3}
+        ::request-history {:wnode      wnode
+                           :topics     request-history-topics
+                           :sym-key-id sym-key-id
+                           :web3       web3}
+        :db                (assoc db :inbox/fetching? true)
+        :dispatch-later    [{:ms fetching-timeout
+                             :dispatch [:inbox/check-fetching now-in-s]}]})))
+  ([should-recover? {:keys [db] :as cofx}]
+   (when should-recover?
```

Comment:
Is this being arity being used? I can't find it in the codebase, probably I am missing something

Reply:
here https://github.com/status-im/status-react/pull/4481/files#diff-9c6dc8fc2461b89e7486af5f1fe7d2f3R459

Reply:
thanks!